### PR TITLE
enhancement : replaced list logic in datasource list to name filter

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISVolumeDatasource_basic (118.27s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	123.947s
```

### CODE REVIEW

**Review by astha-jain — COMMENTED** [review]
Code looks good. Please share the acceptance test output.

File: ibm/data_source_ibm_is_flow_log.go [hunk 1]
```diff
@@ -209,7 +209,9 @@ func dataSourceIBMIsFlowLogRead(context context.Context, d *schema.ResourceData,
 		start := ""
 		allrecs := []vpcv1.FlowLogCollector{}
 		for {
```
Comment (deepaksibm) [inline ibm/data_source_ibm_is_flow_log.go:211]: This for loop can be removed and pagination logic can be removed when name is given as we can directly hit the list api get with name

File: ibm/data_source_ibm_is_image.go [hunk 2]
```diff
@@ -113,7 +113,9 @@ func imageGetByName(d *schema.ResourceData, meta interface{}, name, visibility s
 	start := ""
 	allrecs := []vpcv1.Image{}
 	for {
```
Comment (deepaksibm) [inline ibm/data_source_ibm_is_image.go:115]: same as above, pagination logic can be removed

File: ibm/data_source_ibm_is_virtual_endpoint_gateway.go [hunk 3]
```diff
@@ -122,6 +121,7 @@ func dataSourceIBMISEndpointGatewayRead(
 	allrecs := []vpcv1.EndpointGateway{}
 	for {
```
Comment (deepaksibm) [inline ibm/data_source_ibm_is_virtual_endpoint_gateway.go:122]: i think we can remove for loop in all these changed datasources as only one instance of the resource will be returned with name filter

File: ibm/data_source_ibm_is_flow_log.go [hunk 4]
```diff
@@ -206,25 +206,18 @@ func dataSourceIBMIsFlowLogRead(context context.Context, d *schema.ResourceData,
...
+
+		listOptions := &vpcv1.ListFlowLogCollectorsOptions{
+			Name: &name,
 		}
+
+		flowlogCollectors, response, err := sess.ListFlowLogCollectors(listOptions)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("Error Fetching Flow Logs for VPC %s\n%s", err, response))
+		}
+		allrecs = append(allrecs, flowlogCollectors.FlowLogCollectors...)
```
Comment (deepaksibm) [inline ibm/data_source_ibm_is_flow_log.go:219]: i think we can remove allrecs as well. and add the condition if len(flowlogCollectors) == 0
this applies for all the cases below.

Comment (ujjwal-ibm) [reply ibm/data_source_ibm_is_flow_log.go:219]: `flowlogCollectors` is of type *vpcv1.FlowLogCollectorCollection

Comment (deepaksibm) [reply ibm/data_source_ibm_is_flow_log.go:219]: something like this 
if len(flowlogCollectors.FlowLogCollectors) != 0 {
	flowLogCollector = &flowlogCollectors.FlowLogCollectors[0]
}

**Review by deepaksibm — APPROVED** [review]

**Comment by ujjwal-ibm** [comment]
closing this and will promote to ibm repo

**Pull request closed without merging**